src/Microsoft.Fx.Portability.MetadataReader/ReflectionMetadataDependencyInfo.cs

```diff
@@ -196,5 +197,33 @@ public InvalidPEAssemblyException(Exception inner)
                 : base("Not a valid assembly", inner)
             { }
         }
+
```

Comment:
Have you considered using (ImmutableHashSet<T>)[https://msdn.microsoft.com/en-us/library/dn467171(v=vs.110).aspx]?